workflow/controller/steps.go

```diff
@@ -415,7 +394,29 @@ func (woc *wfOperationCtx) resolveReferences(stepGroup []wfv1.WorkflowStep, scop
 // expandStepGroup looks at each step in a collection of parallel steps, and expands all steps using withItems/withParam
 func (woc *wfOperationCtx) expandStepGroup(sgNodeName string, stepGroup []wfv1.WorkflowStep, stepsCtx *stepsContext) ([]wfv1.WorkflowStep, error) {
 	newStepGroup := make([]wfv1.WorkflowStep, 0)
+	stepTemplateScope := stepsCtx.tmplCtx.GetTemplateScope()
 	for _, step := range stepGroup {
+		childNodeName := fmt.Sprintf("%s.%s", sgNodeName, step.Name)
```

Comment:
This should not be generated independently in two spots. If anything, this should be extracted into a helper function.